tasks/system_probe.py

```diff
@@ -1501,3 +1501,16 @@ def test_microvms(
     ctx.run(
         f"cd ./test/new-e2e && go run ./scenarios/system-probe/main.go --name usama-saqib-test {go_args} --shutdown-period 720",
     )
+
+
+@task
+def generate_git_file_diff(ctx):
+    file_diff = ctx.run(
+        "git diff --name-only $(git merge-base $(inv release.get-release-json-value base_branch) HEAD) HEAD", hide=True
```

Comment:
You can call `inv release.get-release-json-value base_branch` directly as a function rather than using a shell.

Reply:
I can't because ``get_release_json_value`` print his output directly and capturing the python function output is equivalent to running it as is in shell. ref :
```python

@task(help={'key': "Path to the release.json key, separated with double colons, eg. 'last_stable::6'"})
def get_release_json_value(_, key):
    release_json = _load_release_json()

    path = key.split('::')

    for element in path:
        if element not in release_json:
            raise Exit(code=1, message=f"Couldn't find '{key}' in release.json")

        release_json = release_json.get(element)

    print(release_json)
```

Reply:
That makes sense. I think you could extract a common function that could be called, but that can wait for a future PR.